cmd/go-ouroboros-network/chainsync.go

```diff
@@ -0,0 +1,225 @@
+package main
+
+import (
+	"encoding/hex"
+	"fmt"
+	ouroboros "github.com/cloudstruct/go-ouroboros-network"
+	"github.com/cloudstruct/go-ouroboros-network/block"
+	"github.com/cloudstruct/go-ouroboros-network/protocol/blockfetch"
+	"github.com/cloudstruct/go-ouroboros-network/protocol/chainsync"
+	"github.com/cloudstruct/go-ouroboros-network/utils"
+	"os"
+)
+
+type chainSyncState struct {
+	oConn                 *ouroboros.Ouroboros
+	nodeToNode            bool
+	readyForNextBlockChan chan bool
+	byronEpochBaseSlot    uint64
+	byronEpochSlot        uint64
+}
+
+var syncState chainSyncState
+
+// Intersect points (last block of previous era) for each era on testnet/mainnet
```

Comment:
Do you forsee this being a hassle to keep up to date?

Reply:
A little bit, but they are really just here for convenience when testing the library. I can easily update it for a new era (for testnet at least) when I'm adding support for the new era's block format, assuming that the new block format is already in use on the blockchain.